sdc/datatypes/hpat_pandas_series_rolling_types.py

```diff
@@ -0,0 +1,104 @@
+# *****************************************************************************
+# Copyright (c) 2019, Intel Corporation All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#
+#     Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+from numba import cgutils, types
+from numba.datamodel import StructModel
+from numba.extending import (intrinsic, make_attribute_wrapper,
+                             models, register_model)
+from numba.typing.templates import signature
+
+
+class SeriesRollingType(types.Type):
```

Comment:
I wonder if `DataFrame` rolling really should be a different type

Reply:
Do you mean we need to create common type, e.g. `RollingType`? Perhaps you are right. Let me do that.

Reply:
We somehow should separate `SeriesRolling` and `DataFrameRolling` types to be able to overload  their methods separately.

Reply:
But we know data type of member `data` (either series or dataframe). Isn't it enough?
Let's discuss it personally

Reply:
Do you want I make some related changes in this PR?

Reply:
No, i think it's ok for now